crates/wasmprinter/src/lib.rs

```diff
@@ -1459,6 +1473,15 @@ impl Printer {
         Ok(())
     }
 
+    fn print_field_idx(&mut self, state: &State, ty: u32, idx: u32) -> Result<()> {
+        match state.core.field_names.index_to_name.get(&(ty, idx)) {
+            Some(name) => write!(self.result, "${}", name.identifier())?,
+            None if self.name_unnamed => write!(self.result, "$#local{idx}")?,
```

Comment:
It's ok to not implement the `--name-unnamed` feature here if you'd prefer, but otherwise the `#local` here may want to change to `#field` and I think this'll need to update the field-printing-code as well to print this name if a name isn't otherwise listed.

Reply:
Good catch! I had this renamed to field at some point but must've gotten it changed back when shuffling things around.

> I think this'll need to update the field-printing-code as well to print this name if a name isn't otherwise listed.

I think I did that in 4af11e1, but maybe check it matches what you had in mind